dtfs-central-api/src/services/state-machines/utilisation-report/event-handlers/report-uploaded/report-uploaded.event-handler.ts

```diff
@@ -39,7 +42,16 @@ const createFacilityUtilisationDataEntityIfNotExists = async (
   if (entityExists) {
     return null;
   }
-  return FacilityUtilisationDataEntity.createWithoutUtilisationAndFixedFee({ id: facilityId, reportPeriod, requestSource });
+
+  const { utilisation, fixedFee } = await calculateInitialUtilisationAndFixedFee(facilityId);
+
+  return FacilityUtilisationDataEntity.createWithUtilisationAndFixedFee({
+    id: facilityId,
+    reportPeriod,
```

Comment:
When we generate keying data we check that keying data for that report has not already been uploaded by checking that the report period on the utilisation data has not already been calculated (see [here](https://github.com/UK-Export-Finance/dtfs2/blob/b4ef04d7a9956a284bea44ae23f9468ec438122e/dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee-adjustment.ts#L25)).
So imagine if we upload a report with entries for new facilities, it would then throw errors when we generate the keying data later in the flow?
@ChristianMcc do you think we should just use the previous report period here instead? Also just noticed that check we do doesn't explicitly check for the previous report period, it just checks for 'not the current' report period which seems wrong since we want to have protection against reconciling reports out of order ideally.

Reply:
Agreed on using the previous report period instead of current report period, that's what we should do for this.

Re previous report period work, I checked this with both Ian/Soma and Ruth and they have said they will never process them out of order so we don't need to cover that edge case (thankfully) :)